File: FSchool-BE/Domain/Entities/Event.cs
using System.ComponentModel.DataAnnotations;
using Domain.Enums;

namespace Domain.Entities
{
    public class Event
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(200)]
        public string Title { get; set; }

        public DateTime EventDate { get; set; }

        [MaxLength(200)]
        public string Location { get; set; } 

        public string ImageUrl { get; set; }

        public string Description { get; set; }

        public bool IsNews { get; set; }

        public EventStatus Status { get; set; } = EventStatus.Pending;

        public decimal? Budget { get; set; }

        public int? MaxParticipants { get; set; }

        // FK to Club
        public int? ClubId { get; set; }
        [ForeignKey("ClubId")]
        public Club Club { get; set; }

        // FK to Room (for schedule/room conflict checking)
        public int? RoomId { get; set; }
        [ForeignKey("RoomId")]
        public Room Room { get; set; }

        // Navigation Properties
        public ICollection<EventRegistration> Registrations { get; set; }
    }
}

File: FSchool-FE/lib/models/event.dart
class EventModel {
  final String id;
  final String title;
  final String date;
  final String location;
  final String? image;
  final String? description;
  final String? status;
  final int? clubId;
  final double? budget;
  final int? maxParticipants;
  final int registrationCount;

  EventModel({
    required this.id,
    required this.title,
    required this.date,
    required this.location,
    this.image,
    this.description,
    this.status,
    this.clubId,
    this.budget,
    this.maxParticipants,
    this.registrationCount = 0,
  });

  factory EventModel.fromJson(Map<String, dynamic> json) {
    return EventModel(
      id: json['id']?.toString() ?? '',
      title: json['title']?.toString() ?? '',
      date: json['date']?.toString() ?? '',
      location: json['location']?.toString() ?? '',
      image: json['image']?.toString(),
      description: json['description']?.toString(),
      status: json['status']?.toString(),
      clubId: json['clubId'] as int?,
      budget: (json['budget'] as num?)?.toDouble(),
      maxParticipants: json['maxParticipants'] as int?,
      registrationCount: (json['registrationCount'] as num?)?.toInt() ?? 0,
    );
  }
}
